direct/tpm2/read_public_test.go

```diff
@@ -0,0 +1,76 @@
+package tpm2
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/google/go-tpm/direct/structures/tpm"
+	"github.com/google/go-tpm/direct/structures/tpm2b"
+	"github.com/google/go-tpm/direct/structures/tpma"
+	"github.com/google/go-tpm/direct/structures/tpms"
+	"github.com/google/go-tpm/direct/structures/tpmt"
+	"github.com/google/go-tpm/direct/structures/tpmu"
+	"github.com/google/go-tpm/direct/transport/simulator"
+)
+
+func TestReadPublicKey(t *testing.T) {
+
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	// CreatePrimary Struct Params copied from policy_test.go
+	createPrimary := CreatePrimary{
+		PrimaryHandle: tpm.RHOwner,
+		InPublic: tpm2b.Public{
+			PublicArea: tpmt.Public{
+				Type:    tpm.AlgECC,
+				NameAlg: tpm.AlgSHA256,
+				ObjectAttributes: tpma.Object{
+					FixedTPM:            true,
+					FixedParent:         true,
+					SensitiveDataOrigin: true,
+					UserWithAuth:        true,
+					SignEncrypt:         true,
+				},
+				Parameters: tpmu.PublicParms{
+					ECCDetail: &tpms.ECCParms{
+						Scheme: tpmt.ECCScheme{
+							Scheme: tpm.AlgECDSA,
+							Details: tpmu.AsymScheme{
+								ECDSA: &tpms.SigSchemeECDSA{
+									HashAlg: tpm.AlgSHA256,
+								},
+							},
+						},
+						CurveID: tpm.ECCNistP256,
+					},
+				},
+			},
+		},
+	}
```

Comment:
Can we `defer` a `FlushContext` similar to what we do here: https://github.com/google/go-tpm/blob/7cd0fbd6e8fe921362249ce85533e0839dffb68a/direct/tpm2/policy_test.go#L53-L61
Something like:
```go
defer func() { 
 	flush := FlushContext{ 
 		FlushHandle: rsp.ObjectHandle, 
 	} 
 	if err := flush.Execute(thetpm); err != nil { 
 		t.Errorf("could not flush signing key: %v", err) 
 	} 
 }()
```

Reply:
@chrisfenner 
Sorry to make this more convoluted. Common commands like FlushContext take up more lines of code than `go-tpm/tpm2` currently does. Users likely wouldn't want the clutter of management commands like FlushContext taking too much space; they care about the business logic of what they're putting into commands like CreatePrimary or ReadPublic.

Would having the idiom below be easier to read?
```
defer func() {
  if err := FlushContext{FlushHandle: rsp.ObjectHandle}.Execute(thetpm); err != nil {
    t.Errorf("...")
  }
}
```

we could also make wrappers for commands like FlushContext, although I think it seems unnecessary for a small command like FlushContext.
e.g. in a util pkg or something:
```
direct/tpm2/directutil/context
func FlushContext(t transport.TPM, flushHandle handle) error {
  return FlushContext{ FlushHandle: flushHandle}.Execute(t)
}

direct/tpm2/read_public_test.go:
defer func() {
  if err := directutil.FlushContext(thetpm, rsp.ObjectHandle); err != nil {
    t.Errorf("...")
  }
}
```

Reply:
We discussed offline - we don't have to assert that every flush succeeded during the tests (the current tests don't; we just need a FlushContext test that checks it once).

So, I'd like to update my suggestion to just: can we add
```go
defer &FlushContext{rsp.ObjectHandle}.Execute(thetpm))
```
here?

Reply:
I was getting 
`function must be invoked in defer statement` for 

> `defer &FlushContext{rsp.ObjectHandle}.Execute(thetpm))`

So my work around was
```
flushContext := FlushContext{FlushHandle: rspCP.ObjectHandle}
defer flushContext.Execute(thetpm)
```
which seems to work just as well

Reply:
Can you try `defer (&FlushContext{rsp.ObjectHandle}).Execute(thetpm))`

Reply:
The same error occurs.

Reply:
I'm fine with the 2 line version here, and if we think of a neat idiom later it's easy to change.